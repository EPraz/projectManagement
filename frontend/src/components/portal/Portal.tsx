import ReactDOM from "react-dom";
import { PageContent } from "../layout/Layout.styles";
import { PortalProps } from "../../types";

const Portal = ({ children }: PortalProps) => {
  return ReactDOM.createPortal(
    <PageContent>{children}</PageContent>,
    document.body
  );
};

export default Portal;
