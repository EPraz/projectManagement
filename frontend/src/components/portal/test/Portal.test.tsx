import { render, screen } from "@testing-library/react";
import Portal from "../Portal";

describe("Portal Component", () => {
  it("match snapshot", async () => {
    const { asFragment } = render(
      <Portal>
        <div data-testid="portal-content">Portal Content</div>
      </Portal>
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("renders children into document.body", () => {
    render(
      <Portal>
        <div data-testid="portal-content">Portal Content</div>
      </Portal>
    );

    const content = screen.getByTestId("portal-content");
    expect(content).toBeInTheDocument();
    expect(document.body).toContainElement(content);
  });
});
