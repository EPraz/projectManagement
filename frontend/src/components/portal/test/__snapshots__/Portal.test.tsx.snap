// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal Component match snapshot 1`] = `[Function]`;
